Assets/StreamingAssets/CSharp/DialogBoxFunctions.cs

```diff
@@ -0,0 +1,1212 @@
+                Character copy = character;
+                GameObject copyGo = job;
+                copy.InterruptState();
+                GameObject.Destroy(copyGo);
+            });
+            delete.GetComponent<RectTransform>().pivot = new Vector2(1, 0.5f);
+            AspectRatioFitter ratio = delete.gameObject.AddComponent<AspectRatioFitter>();
+            ratio.aspectMode = AspectRatioFitter.AspectMode.HeightControlsWidth;
+            GameObject.Destroy(delete.GetComponent<LayoutElement>());
+
+            DialogBoxJobListItem item = job.AddComponent<DialogBoxJobListItem>();
+            item.box = this;
+            item.character = character;
+            item.index = i;
+            i++;
+        }
+
+        scroll.scrollSensitivity = content.transform.childCount / 2;
+
+        GameObject bottomBar = GetFluidHorizontalBaseElement("Bottom", true, true, allocatedHeight: 40);
+        HorizontalLayoutGroup layout = bottomBar.GetComponent<HorizontalLayoutGroup>();
+        layout.padding = new RectOffset(30, 30, 0, 30);
+        layout.spacing = 10;
+        layout.transform.SetParent(element.transform);
+        layout.childForceExpandHeight = false;
+        layout.childAlignment = TextAnchor.LowerCenter;
+
+        Button okay = CreateButton("okay");
+        okay.transform.SetParent(bottomBar.transform);
+        okay.onClick.AddListener(() => {
+            DialogBoxManager.FindInstance().SoftCloseTopDialog();
+        });
+
+
+        return element;
+    }
+}
+
+public class TradeItemElement
+{
+    public Text playerStock;
+    public InputField tradeAmount;
+    public Text tradeStock;
+    public TradeItem item;
+
+    public void UpdateInterface()
+    {
+        playerStock.text = (item.PlayerStock + item.TradeAmount).ToString();
+        tradeStock.text = (item.TraderStock - item.TradeAmount).ToString();
+        tradeAmount.text = item.TradeAmount.ToString();
+    }
+}
+
+public class DialogBoxTrade : BaseDialogBox
+{
+    private GameObject content;
+    private Text playerCurrencyBalance;
+    private Text traderCurrencyBalance;
+    private Text tradeCost;
+    private Button acceptButton;
+    public Trade currentTrade;
+
+    public void BuildHeader()
+    {
+        float tradeAmount = currentTrade.TradeCurrencyBalanceForPlayer;
+        playerCurrencyBalance.text = string.Format(
+            "{0:N2} {1}", 
+            currentTrade.Player.Currency.Balance + currentTrade.TradeCurrencyBalanceForPlayer, 
+            currentTrade.Player.Currency.ShortName);
+        traderCurrencyBalance.text = string.Format(
+            "{0:N2} {1}", 
+            currentTrade.Trader.Currency.Balance - currentTrade.TradeCurrencyBalanceForPlayer, 
+            currentTrade.Trader.Currency.ShortName);
+        tradeCost.text = tradeAmount.ToString("N2");
+        acceptButton.interactable = currentTrade.IsValid();
+    }
+
+    public override GameObject InitializeElement()
+    {
+        if (!callerData.ContainsKey("Trade"))
```

Comment:
Should use `TryGetValue` instead

Reply:
I don’t think so, cause it needs ro retrieve Trade object by casting the inner object else it would look like;

object value;
Trade trade;
If (tryGetValye()==false) ...
trade = (Trade)value

This is honestly uglier than the check and ger